Logic/logic_manager.cpp

```diff
@@ -8,46 +10,116 @@
+bool p2p::PARSER_LOGS_ON;
+bool p2p::PARSER_RESPONSE_LOGS_ON;
+bool p2p::USERS_SERVER_LOGS_ON;
+bool p2p::ORDERS_FOR_BUY_LOGS_ON;
+bool p2p::ORDERS_FOR_SELL_LOGS_ON;
+bool p2p::SELL_BY_COIN_LOGS_ON;
+bool p2p::MARKET_RATES_LOGS_ON;
+bool p2p::CHAINS_LOGS_ON;
+
 namespace p2p {
 
-// Запускаем в отдельном потоке из Main::run
-void raise_parser_connection(std::string parser_connection_ip,
-                             std::string parser_connection_port) {
-  // Поднимаем клиента для подключения к парсеру:
-  boost::asio::io_context client_io_context;
-  ip::tcp::resolver resolver(client_io_context);
-  auto endpoints =
-      resolver.resolve(parser_connection_ip, parser_connection_port);
-  ParserConnectionClient client(client_io_context, endpoints);
-  client_io_context.run();
+void raise_parser_connection_and_users_server() {
+  boost::asio::io_context global_io_context;
+  ip::tcp::resolver resolver(global_io_context);
+  auto endpoints = resolver.resolve(PARSER_IP, std::to_string(PARSER_PORT));
+  ParserConnectionClient client(global_io_context, endpoints);
+  auto users_server = UsersConnectionsServer(global_io_context);
+  users_server.listen(USERS_SERVER_PORT);
+  global_io_context.run();
 }
 
-// Запускаем в отдельном потоке из Main::run
-void raise_users_server(uint16_t users_server_port) {
-  // Поднимаем сервер для подключений клиентов:
-  auto users_server = UsersConnectionsServer();
-  users_server.listen(users_server_port);
-  users_server.run();
+void to_analysis() {
+  std::string response;
+  std::string last_response;
+  while (true) {
+    bool has_new_responses = false;
+    while (parsers_responses.try_dequeue(response)) {
+      last_response = response;
+      has_new_responses = true;
+    }
+    if (has_new_responses) {
+      if (PARSER_RESPONSE_LOGS_ON) {
+        std::cout << "Last response: " << last_response << std::endl;
+      }
+
+      Orders orders;
+      MarketRates market_rates;
+      if (!last_response.empty() && last_response[0] == '{') {
+        unpack_json(last_response, orders, market_rates);
+      } else {
+        if (PARSER_RESPONSE_LOGS_ON) {
+          std::cout << "Received empty response.\n";
+        }
+        continue;
+      }
+
+      Analysis analysis;
+      Chains chains;
+      analysis.analyze(chains, orders, market_rates);
+
+      if (CHAINS_LOGS_ON) {
+        std::cout << "Chains size: " << chains.list.size() << std::endl;
+        std::cout << "Chains:" << std::endl;
+        for (auto &chain : chains.list) {
+          std::cout << "Buy: " << chain.buy.coin2 << ", " << chain.change.first
+                    << " -> " << chain.change.second
+                    << ", Sell: " << chain.sell.coin2
+                    << ", Spread: " << chain.spread << "%" << std::endl;
+        }
+      }
+
+      up_to_date_version.set(pack_json(chains));
+
+    } else {
+      std::this_thread::sleep_for(
+          std::chrono::milliseconds(100)); // Пауза на 100 миллисекунд
```

Comment:
why don't you use `deque/try_deque_until/try_deque_for` from moodcamel instead of `try_dequeue`? So you could've removed this sleep